backend/api/views.py

```diff
@@ -0,0 +1,151 @@
+        SAMPLE_RESPONSE:
+        [
+            {
+                "id": 2,
+                "name": "Office 2",
+                "street": "Street 2",
+                "postal_code": "123123123",
+                "city": "Moscow",
+                "monthly_rent": "1.12",
+                "headquarter": false,
+                "company": 1
+            },
+            ...
+        ]
+        '''
+        if request.GET['company']:
+            offices = Office.objects.filter(company__name=request.GET['company']).all()  # noqa
+        else:
+            offices = Office.objects.all()
+
+        serializer = OfficeSerializer(offices, many=True)
+        return Response(serializer.data)
+
+
+class UpdateHeadquarter(APIView):
+    def put(self, request, pk, format=None):
+        '''
+        REQUEST_METHOD: PUT,
+        ENDPOINT: /api/v1/company/<company_id>/
+        REQUEST_PARAM:
+        {
+            "office_id": 1
+        }
+        SAMPLE_RESPONSE:
+        [
+            {
+                "id": 3,
+                "name": "Office 2",
+                "street": "Street 2",
+                "postal_code": "10995",
+                "city": "city3",
+                "monthly_rent": "1.20",
+                "headquarter": true,
+                "company": 1
+            }
+        ]
+        '''
+        Office.objects.filter(company__pk=pk).update(headquarter=False)
+        Office.objects.filter(id=request.data['office_id']).update(headquarter=True)  # noqa
+        query_set = Office.objects.filter(company__pk=pk, headquarter=True).all()  # noqa
+        serializer = OfficeSerializer(query_set, many=True)
+
+        return Response(serializer.data)
+
+
+class CreateCompanyOffice(APIView):
+    def post(self, request):
+        '''
+        METHOD: POST
+        ENDPOINT: /api/v1/add_company/
+        REQEST PARAM:
+        {
+            "name": "my another company",
+            "offices": [
+                {
+                    "name": "Office 23",
+                    "street": "Street 23",
+                    "postal_code": "102323",
+                    "city": "city23",
+                    "monthly_rent": "1.20",
+                    "headquarter": true
+                },
+                {
+                    "name": "Office 23",
+                    "street": "Street 23",
+                    "postal_code": "102323",
+                    "city": "city23",
+                    "monthly_rent": "1.20",
+                    "headquarter": false
+                }
+            ]
+        }
+        SAMPLE_RESPONSE:
+        [
+            {
+                "id": 8,
+                "name": "my company1",
+                "headquarter_office": {
+                    "street": "Street 23",
+                    "city": "city23",
+                    "postal_code": "102323"
+                }
+            }
+        ]
+        '''
+        company = Company.objects.create(name=request.data['name'])
```

Comment:
Bad here, It can be easily done by serializer. we don't have to write much logic and hard code in view what we can do easily in serializer.

Reply:
I see. Actually, I am more talented in Flask rather than Django. This made me to have this kind of habit. Will figure it out in the next PR.